proto/substrait/relations.proto

```diff
@@ -568,11 +568,86 @@ message Expression {
     oneof root_type {
       Expression expression = 3;
       RootReference root_reference = 4;
+      OuterReference outer_reference = 5;
     }
 
     // Singleton that expresses this FieldReference is rooted off the root incoming record type
     message RootReference {}
+    message OuterReference {
+      // number of subquery boundaries to traverse up for this field's reference
+      //
+      // an outer reference is a root reference for the outer relation's
+      // subquery
+      //
+      // This value must be >= 1
+      uint32 steps_out = 1;
+    }
+  }
 
+  message Subquery {
+    // Subquery relation expression
+    oneof subquery_type {
+      Scalar scalar = 1;
+      SetPredicate exists_predicate = 2;
```

Comment:
the name of this union item doesn't match the type.

Reply:
Fixed.